src/aktualizr_info/aktualizr_info_config.cc

```diff
@@ -0,0 +1,89 @@
+#include "aktualizr_info_config.h"
+
+#include "utilities/config_utils.h"
+
+AktualizrInfoConfig::AktualizrInfoConfig(const boost::program_options::variables_map& cmd) {
+  // Redundantly check and set the loglevel from the commandline prematurely so
+  // that it is taken account while processing the config.
+  if (cmd.count("loglevel") != 0) {
+    logger.loglevel = cmd["loglevel"].as<int>();
+    logger_set_threshold(logger);
+    loglevel_from_cmdline = true;
+  }
+
+  if (cmd.count("config") > 0) {
+    updateFromDirs(cmd["config"].as<std::vector<boost::filesystem::path>>());
+  } else {
+    updateFromDirs(config_dirs_);
+  }
+  updateFromCommandLine(cmd);
+  postUpdateValues();
+}
+
+AktualizrInfoConfig::AktualizrInfoConfig(const boost::filesystem::path& filename) {
+  updateFromToml(filename);
+  postUpdateValues();
+}
+
+void AktualizrInfoConfig::postUpdateValues() { logger_set_threshold(logger); }
+
+void AktualizrInfoConfig::updateFromDirs(const std::vector<boost::filesystem::path>& configs) {
```

Comment:
We might want to factor this kind of methods to some common utilities or (maybe!) base class before it goes out of hand.

Not critical though

Reply:
Agreed, there is probably an even smarter way of solving this. I'll ponder that, but for now, I think this is at least an improvement to what was there before.